packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -430,33 +430,35 @@ export class DataStores implements IDisposable {
         );
     }
 
-    public async getDataStore(id: string, wait: boolean, viaHandle: boolean): Promise<FluidDataStoreContext> {
-        const context = await this.contexts.getBoundOrRemoted(id, wait);
+    public async getDataStore(id: string, headerData: RuntimeHeaderData): Promise<FluidDataStoreContext> {
+        const context = await this.contexts.getBoundOrRemoted(id, headerData.wait);
         const request = { url: id };
         if (context === undefined) {
             // The requested data store does not exits. Throw a 404 response exception.
             throw responseToException(create404Response(request), request);
         }
 
         if (context.tombstoned) {
+            const shouldFail = this.throwOnTombstoneUsage && !headerData.allowTombstone;
+
             // The requested data store is removed by gc. Create a 404 gc response exception.
             const error = responseToException(createResponseError(404, "Datastore removed by gc", request), request);
-            // Note: if a user writes a request to look like it's viaHandle, we will also send this telemetry event
-            const event = {
-                eventName: "GC_Tombstone_DataStore_Requested",
-                url: request.url,
-                viaHandle,
-            };
+
+            // Note: if a user writes a request to look like it's viaHandle,
+            // it will be indistinguishable in telemetry from an actual handle
             sendGCTombstoneEvent(
-                this.mc,
-                event,
+                this.mc.logger,
+                {
+                    eventName: "GC_Tombstone_DataStore_Requested",
+                    viaHandle: headerData.viaHandle,
```

Comment:
Let's just log `...headerData` so we can also tell if allowTombstone as set or not.